python_modules/libraries/dagster-k8s/dagster_k8s/executor.py

```diff
@@ -264,6 +280,24 @@ def _get_k8s_step_job_name(self, step_handler_context: StepHandlerContext):
 
         return f"dagster-step-{name_key}"
 
+    def _detect_current_name_and_uid(
+        self,
+    ) -> Optional[tuple[str, str]]:
```

Comment:
right now this is called once per launch_step call - but it never changes for a given process right? throw a cached_method in here?